backend/src/main/java/fk/prof/backend/model/assignment/impl/SimultaneousWorkAssignmentCounterImpl.java

```diff
@@ -0,0 +1,36 @@
+package fk.prof.backend.model.assignment.impl;
+
+import fk.prof.backend.model.assignment.SimultaneousWorkAssignmentCounter;
+import fk.prof.backend.proto.BackendDTO;
+import recording.Recorder;
+
+/**
+ * No operation of this class is thread-safe
+ * Right now, only invoked from backend daemon thread, so we are good
+ */
+public class SimultaneousWorkAssignmentCounterImpl implements SimultaneousWorkAssignmentCounter {
+  private final int maxAllowedConcurrentWorkAssignmentSlots;
+  private int currentlyAcquiredWorkAssignmentSlots = 0;
+
+  public SimultaneousWorkAssignmentCounterImpl(int maxAllowedConcurrentWorkAssignmentSlots) {
+    this.maxAllowedConcurrentWorkAssignmentSlots = maxAllowedConcurrentWorkAssignmentSlots;
+  }
+
+  /**
+   * TODO: Should be refactored to take process group and work prototype as input and accordingly determine slots
+   * @return
+   */
+  public int acquireSlots(Recorder.ProcessGroup processGroup, BackendDTO.WorkProfile workProfile) {
```

Comment:
Fixed in commit https://github.com/Flipkart/fk-prof/commit/14c6b7773d38b356a1a8daa5cf5a98e8ef5862ab